feat: Add OpenHands Flutter development tools and automation

jawsbaek opened the pull request:
## OpenHands 도입을 위한 세팅
https://docs.all-hands.dev/usage/prompting/prompting-best-practices

## 📋 Summary

This PR adds comprehensive Flutter development tools and automation scripts to streamline the development workflow. The `.openhands` directory contains scripts for environment setup, code quality validation, and integrated development commands.

## 🚀 What's Added

### New Files
- **`.openhands/setup.sh`** - Automated development environment setup script
- **`.openhands/pre-commit.sh`** - Pre-commit validation script for code quality
- **`.openhands/dev-tools.sh`** - Integrated Flutter development commands
- **`.openhands/README.md`** - Comprehensive documentation and usage guide
- **`.pre-commit-config.yaml`** - Pre-commit hooks configuration

## ✨ Features

### 🔧 Development Environment Setup (`setup.sh`)
- Automatic Flutter SDK installation and validation
- Project dependencies installation (`flutter pub get`)
- Pre-commit tools installation
- Git hooks configuration
- Flutter Doctor execution
- Project validation (analysis + tests)

### 🛡️ Code Quality Automation (`pre-commit.sh`)
- **Code formatting** - Automatic Dart code formatting with `dart format`
- **Code analysis** - Static analysis with `flutter analyze`
- **Test execution** - Automated testing for changed code
- **Dependency validation** - Checks for `pubspec.yaml` changes
- **File size monitoring** - Warns about large files (>1MB)

### 🛠️ Integrated Development Tools (`dev-tools.sh`)
Available commands:
- `setup` - Run development environment setup
- `clean` - Clean project and dependencies
- `build` - Build app for different platforms (web/android/ios)
- `test` - Execute tests
- `analyze` - Run code analysis
- `format` - Format code
- `run-web` - Run app in web browser
- `run-mobile` - Run app on mobile device
- `doctor` - Run Flutter Doctor
- `deps` - Update dependencies
- `gen` - Generate code (if build_runner is configured)

### 📋 Pre-commit Hooks Configuration
- **Flutter/Dart specific hooks** for formatting and analysis
- **General file validation** (YAML, JSON, file sizes, etc.)
- **Conventional commit message validation**
- **Automatic code formatting** on commit

## 🌐 Web Development Support

Optimized for OpenHands environment with proper port configuration:
```bash
# Run on port 12000 (default)
./.openhands/dev-tools.sh run-web

# Run on port 12001
./.openhands/dev-tools.sh run-web 12001
```

## 🚦 Usage

### Quick Start
```bash
# Make scripts executable
chmod +x .openhands/*.sh

# Set up development environment
./.openhands/setup.sh

# Install pre-commit hooks
pip install pre-commit
pre-commit install
```

### Daily Development
```bash
# Format and analyze code
./.openhands/dev-tools.sh format
./.openhands/dev-tools.sh analyze

# Run tests
./.openhands/dev-tools.sh test

# Run web app
./.openhands/dev-tools.sh run-web
```

## 🔍 Code Quality Improvements

- **Automated formatting** ensures consistent code style
- **Static analysis** catches potential issues early
- **Pre-commit validation** prevents problematic commits
- **Conventional commits** improve commit message quality
- **Comprehensive testing** maintains code reliability

## 📚 Documentation

Complete documentation is provided in `.openhands/README.md` including:
- Detailed script descriptions
- Usage examples
- Troubleshooting guide
- Customization instructions
- Commit message conventions

## 🧪 Testing

All scripts have been tested and include:
- Error handling and validation
- Colored output for better UX
- Cross-platform compatibility (Linux/macOS)
- Proper exit codes for CI/CD integration

## 🔄 Workflow Integration

These tools integrate seamlessly with:
- **Git workflows** - Pre-commit hooks and conventional commits
- **CI/CD pipelines** - Proper exit codes and validation
- **IDE integration** - Works with any editor/IDE
- **Team development** - Consistent environment across developers

## 📝 Commit Message Convention

This PR follows conventional commit format:
- `feat:` for new features
- `fix:` for bug fixes
- `docs:` for documentation
- `style:` for formatting
- `refactor:` for code refactoring
- `test:` for testing
- `chore:` for maintenance

## 🎯 Benefits

1. **Faster onboarding** - New developers can set up environment with one command
2. **Consistent code quality** - Automated formatting and analysis
3. **Reduced errors** - Pre-commit validation catches issues early
4. **Improved productivity** - Integrated commands reduce context switching
5. **Better collaboration** - Standardized workflows and commit messages

## 🔧 Future Enhancements

The modular design allows for easy extension:
- Additional code quality checks
- Custom build configurations
- Integration with external tools
- Platform-specific optimizations

---

**Ready for review!** 🚀 This PR significantly improves the Flutter development experience with comprehensive automation and quality tools.

openhands-agent committed:
feat: add OpenHands Flutter development tools and automation

- Add .openhands/ directory with comprehensive Flutter development scripts
- Add setup.sh for automated development environment configuration
- Add pre-commit.sh for code quality validation before commits
- Add dev-tools.sh for integrated Flutter development commands
- Add .pre-commit-config.yaml for automated code quality checks
- Include comprehensive documentation in .openhands/README.md

Features:
- Automated Flutter SDK installation and validation
- Code formatting, analysis, and testing automation
- Pre-commit hooks for maintaining code quality
- Integrated development commands (build, test, run, etc.)
- Support for web and mobile development workflows
- Conventional commit message validation

choseongmin1997 commented on .openhands/README.md:
````diff
@@ -0,0 +1,221 @@
+
+**사용법:**
+```bash
+./.openhands/dev-tools.sh [명령어]
````
명령어를 실행해보니 루트 디렉토리에서 실행해야만 되더라구요..! 간혹 .openhands 디렉토리에 들어가서 실행하는 경우를 경고하는 내용이 추가되면 어떨까요?

choseongmin1997 reviewed (approved):
`./dev-tools.sh gen` 명령어를 제외한 나머지는 리드미대로 테스트해보니 잘 되었습니다!
이후 PR 생성 시 pre-commit 검사를 했는지 여부 등을 체크박스로 확인해보는 것도 좋을 것 같아요

jawsbaek commented:
> `./dev-tools.sh gen` 명령어를 제외한 나머지는 리드미대로 테스트해보니 잘 되었습니다! 이후 PR 생성 시 pre-commit 검사를 했는지 여부 등을 체크박스로 확인해보는 것도 좋을 것 같아요

프리 커밋 실행을 깃액션으로 자동화 걸어둘게요

openhands-agent committed:
feat: Add directory validation to prevent execution from wrong location

- Add check_project_root() function to all three scripts
- Validate that scripts are run from Flutter project root directory
- Provide helpful error messages with correct usage instructions
- Add specific guidance for users who might run scripts from .openhands directory
- Update README.md with important execution location requirements
- Add troubleshooting section for directory-related errors

Addresses reviewer feedback about ensuring scripts are executed from the correct location.
